src/contrib/cluster/Akka.DistributedData/DeltaPropagationSelector.cs

```diff
@@ -105,19 +104,17 @@ public void Delete(string key)
                             var cacheKey = (key, fromSeqNr, toSeqNr);
                             if (!cache.TryGetValue(cacheKey, out var deltaGroup))
                             {
-                                using (var e = deltaEntriesAfterJ.Values.GetEnumerator())
+                                deltaGroup = deltaEntriesAfterJ.Values.Aggregate((d1, d2) =>
```

Comment:
Replaced the suspicious looking code block with one that makes a lot more sense.
This is the scala code for reference:
```scala
val group = deltaEntriesAfterJ.valuesIterator.reduceLeft { (d1, d2) =>
  val merged = d2 match {
    case NoDeltaPlaceholder => NoDeltaPlaceholder
    case _                  =>
      // this is fine also if d1 is a NoDeltaPlaceholder
      d1.merge(d2.asInstanceOf[d1.T])
  }
  merged match {
    case s: ReplicatedDeltaSize if s.deltaSize >= maxDeltaSize =>
      // discard too large deltas
      NoDeltaPlaceholder
    case _ => merged
  }
}
```
`reduceLeft` works exactly the same as LINQ `Aggregate`